### PR TITLE
Add authorization to jobs module

### PR DESCRIPTION
**Pull request by ninov**
This change will allow users to require authentication via HTTP Basic Authentication prior to accessing the /@jobs page.

It introduces the following config options:
- `jobs.auth` (bool, default=`false`)  Is authentication required?
- `jobs.auth.user` (string) Required username
- `jobs.auth.pass` (string) Required password (or its SHA-256-hash, if `jobs.auth.sha256` is `true`)
- `jobs.auth.sha256` (bool, default=`false`) Is the password SHA-256-hashed? You probably don't want your passwords in clear text in config files.

If no user and pass is specified in the config, access will always be denied.
If authorization is enabled, local IP checking will be disabled.

_This feature was requested at [the `jobs` documentation page](http://revel.github.io/manual/jobs.html#future-areas-for-development), so I felt free to implement it._

### CODE REVIEW

**Comment by notzippy** [comment]
@ninov Are you still available to resolve the conflicting files ? Also can you look at this issue #21 and see if you can implement a fix with your code ?

thanks

**Comment by ninov** [comment]
Hi, I merged the conflicting files, it should work now.

File: jobs/app/controllers/status.go [hunk 1]
```diff
@@ -1,33 +1,84 @@
...
-		return c.Forbidden("%s is not local", remoteAddress)
-	}
+
 	entries := jobs.MainCron.Entries()
 	return c.Render(entries)
 }
 
+func (c *Jobs) unauthorized() revel.Result {
+	c.Response.Status = http.StatusUnauthorized
+	c.Response.Out.Header().Set("WWW-Authenticate", "Basic realm=\"revel jobs\"")
```
Comment (pedromorgan) [inline jobs/app/controllers/status.go]: So as I under this is a pop.. and maybe even an .htacess problem from younder year

Snag in my head is that its not "verbose enough" as popup dialog will say "revel jobs"

On a company system etc this is confused..
So maybe we can add
```
The www server at {fill.example.com} required some crendentials
```

**Comment by pedromorgan** [comment]
Parsing an existing .htacess and import to routes it would be cool

File: jobs/app/controllers/status.go [hunk 2]
```diff
@@ -1,33 +1,84 @@
...
-	"strings"
-
-	"github.com/revel/cron"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
 	"github.com/revel/modules/jobs/app/jobs"
 	"github.com/revel/revel"
+	"github.com/robfig/cron"
```
Comment (notzippy) [inline jobs/app/controllers/status.go]: Can you change this to be github.com/revel/cron ?

File: jobs/app/controllers/status.go [hunk 3]
```diff
@@ -1,33 +1,84 @@
...
+		}
+
+	} else {
+		if revel.Config.BoolDefault("jobs.acceptproxyaddress", false) {
+			if proxiedAddress, isProxied := c.Request.Header["X-Forwarded-For"]; isProxied {
+				remoteAddress = proxiedAddress[0]
+			}
+		}
+		if !strings.HasPrefix(remoteAddress, "127.0.0.1") &&
+      !strings.HasPrefix(remoteAddress, "::1") &&
```
Comment (notzippy) [inline jobs/app/controllers/status.go]: Is there some weird tabs vs spaces which is messing up this indent ? Can you fix this ?

File: jobs/app/controllers/status.go (hunk 1 above)
Comment (notzippy) [inline jobs/app/controllers/status.go]: maybe something like

    c.Response.Out.Header().Set("WWW-Authenticate", "Basic realm=\"" +
        revel.Config.StringDefault("jobs.realm", "Revel Jobs")+"\"")

**Comment by pedromorgan** [comment]
Snag I got is that it dont say domain.. Im my case a test server on lan in RPI and my machine..

TPI had apache, prod server had nginx where "htavcess" is a knda plugin.. me thinks. not xPert

The popup message need to show a message of which domain it is and stuff..

But thew ragain, the revel config does not include the "site etc".. sigh !!

**Pull request closed without merging**